device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java

```diff
@@ -331,270 +315,10 @@ public void completeFileUpload(FileUploadCompletionNotification notification) th
-     *	      in case of MQTT and AMQP protocols, this option specifies the interval in milliseconds
-     *	      between spawning a thread that dequeues a message from the SDK's queue of received messages.
-     *
-     *	    - <b>SetMaxMessagesSentPerThread</b> - this option is applicable to all protocols.
-     *	      This option specifies how many messages a given send thread should attempt to send before exiting.
-     *	      This option can be used in conjunction with "SetSendInterval" to control the how frequently and in what
-     *	      batch size messages are sent. By default, this client sends 10 messages per send thread, and spawns
-     *	      a send thread every 10 milliseconds. This gives a theoretical throughput of 1000 messages per second.
-     *
-     *      - <b>SetSASTokenExpiryTime</b> - this option is applicable for HTTP/
-     *         AMQP/MQTT. This option specifies the interval in seconds after which
-     *         SASToken expires. If the transport is already open then setting this
-     *         option will restart the transport with the updated expiry time, and
-     *         will use that expiry time length for all subsequently generated sas tokens.
-     *         The value is expected to be of type {@code long}.
-     *
-     *      - <b>SetHttpsReadTimeout</b> - this option is applicable for HTTPS.
-     *         This option specifies the read timeout in milliseconds per https request
-     *         made by this client. By default, this value is 4 minutes.
-     *         The value is expected to be of type {@code int}.
-     *
-     *      - <b>SetHttpsConnectTimeout</b> - this option is applicable for HTTPS.
-     *         This option specifies the connect timeout in milliseconds per https request
-     *         made by this client. By default, this value is 0 (no connect timeout).
-     *         The value is expected to be of type {@code int}.
-     *
-     *      - <b>SetAmqpOpenAuthenticationSessionTimeout</b> - this option is applicable for AMQP with SAS token authentication.
-     *         This option specifies the timeout in seconds to wait to open the authentication session.
-     *         By default, this value is 20 seconds.
-     *         The value is expected to be of type {@code int}.
-     *
-     *      - <b>SetAmqpOpenDeviceSessionsTimeout</b> - this option is applicable for AMQP.
-     *         This option specifies the timeout in seconds to open the device sessions.
-     *         By default, this value is 60 seconds.
-     *         The value is expected to be of type {@code int}.
-     *
-     * @param optionName the option name to modify
-     * @param value an object of the appropriate type for the option's value
-     * @throws IllegalArgumentException if the provided optionName is null
-     */
-    public void setOption(String optionName, Object value) throws IllegalArgumentException
```

Comment:
DeviceClient no longer does anything different from ModuleClient in the setOption flow, so no need to override it from InternalClient anymore